Sources/MongoSwift/BSON/BSONDecoder.swift

```diff
@@ -192,25 +192,25 @@ internal struct _BSONDecodingStorage {
 /// Extend _BSONDecoder to add methods for "unboxing" values as various types.
 extension _BSONDecoder {
 
-    fileprivate func unboxBSONValue<T: BSONValue>(_ value: BSONValue?, as type: T.Type) throws -> T? {
+    fileprivate func unboxBSONValue<T: BSONValue>(_ value: BSONValue, as type: T.Type) throws -> T? {
```

Comment:
is there a reason this function returns an optional? I know I'm the one who wrote it that way but I'm not sure why...

Reply:
Ah good catch. I missed these because I think I only searched for instances of `nil` and `BSONValue?`.

This is essentially trying to return a `BSONValue` for which it can no longer return `nil`. If you look at where this function, it seems like they are trying to catch cases where it returns `nil` as well.

Reply:
In other words, I think this can return a non-optional now that we have `NSNull`, we're just going to have to change the places at which we call this function to check for `NSNull` instead of checking for `nil` and throwing.

Reply:
Done.

I won't comment "Done." on the others to avoid potential notification spam, already feel bad about "See answer to question on unboxBSONValue()." 😅.

But they are all done, so feel free to resolve them all.